labelbox/schema/project.py

```diff
@@ -424,6 +443,99 @@ def setup(self, labeling_frontend, labeling_frontend_options):
         timestamp = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
         self.update(setup_complete=timestamp)
 
+    def queue(self, data_rows: List):
+        """Add DataRows to the Project queue"""
+
+        if not self._is_batch_mode():
+            warnings.warn("Project not in Batch mode")
+
+        method = "submitBatchOfDataRows"
+        return self._post_batch(method, data_rows)
+
+    def dequeue(self, data_rows: List):
+
+        if not self._is_batch_mode():
+            warnings.warn("Project not in Batch mode")
+
+        method = "removeBatchOfDataRows"
+        return self._post_batch(method, data_rows)
+
+    def _post_batch(self, method, data_rows):
+        """Create """
+
+        ids = [dr.uid for dr in data_rows]
+
+        if len(ids) > MAX_BATCH_SIZE:
+            raise ValueError("1000 Max DataRows at a time")
+
+        query = """mutation %sPyApi($projectId: ID!, $dataRowIds: [ID!]!) {
+              project(where: {id: $projectId}) {
+                %s(data: {dataRowIds: $dataRowIds}) {
+                  dataRows {
+                    dataRowId
+                    error
+                  }
+                }
+              }
+            }         
+        """ % method
+
+        res = self.client.execute(query, {
+            "projectId": self.uid,
+            "dataRowIds": ids
+        })["project"][method]["dataRows"]
+
+        # TODO: figure out error messaging
+        if len(data_rows) == len(res):
+            raise ValueError("No dataRows were submitted successfully")
+
+        if len(data_rows) > 0:
+            warnings.warn("Some Data Rows were not submitted successfully")
+
+        return res
+
+    def _update_queue_mode(self, mode: QueueMode):
+
+        if mode == QueueMode.Batch:
+            status = "ENABLED"
+        elif mode == QueueMode:
+            status = "DISABLED"
+        else:
+            raise ValueError(
+                "Must provide either `BATCH` or `DATASET` as a mode"
+            )
+
+        query_str = """mutation %s($projectId: ID!, $status: TagSetStatusInput!) {
+              project(where: {id: $projectId}) {
+                 setTagSetStatus(input: {tagSetStatus: $status}) {
+                    id
+                    tagSetStatus
+                    __typename
+                }
+                __typename
+            }
+        }       
+        """ % "setTagSetStatusPyApi"
+
+        self.client.execute(query_str, {
+            'projectId': self.uid,
+            'status': status
+        })
+        self.tag_set_status = status
```

Comment:
tag_set_status is not going to be defined until this point. On Project init we should fetch the existing status.